app/ui-react/syndesis/src/modules/integrations/components/editor/AddStepPage.tsx

```diff
@@ -42,72 +63,176 @@ export interface IAddStepPageProps extends IGetStepHrefs {
+                        )}
+                        showDialog={this.state.showDeleteDialog}
+                        onCancel={this.closeDeleteDialog}
+                        onConfirm={() => {
+                          this.handleDeleteConfirm();
+
+                          /**
+                           * Remove the step from the integration flow
+                           * and receive a copy of the new integration.
+                           */
+                          const newInt = removeStepFromFlow(
+                            integration,
+                            flowId,
+                            this.state.position!
+                          );
 
-              return (
-                <>
-                  <PageTitle title={t('integrations:editor:saveOrAddStep')} />
-                  <IntegrationEditorLayout
-                    title={t('integrations:editor:addToIntegration')}
-                    description={t('integrations:editor:addStepDescription')}
-                    content={
-                      <IntegrationEditorStepAdder
-                        steps={getSteps(integration, flowId)}
-                        addDataMapperStepHref={position =>
-                          this.props.getAddMapperStepHref(
-                            position,
-                            { flowId },
-                            { integration }
-                          )
-                        }
-                        addStepHref={position =>
-                          this.props.getEditAddStepHref(
-                            position,
-                            { flowId },
-                            { integration }
-                          )
-                        }
-                        configureStepHref={(position: number, step: Step) =>
-                          getStepHref(
-                            step,
-                            { flowId, position: `${position}` },
-                            { integration },
-                            this.props
-                          )
-                        }
-                        flowId={flowId}
-                        integration={integration}
-                        deleteAction={deleteAction}
+                          /**
+                           * Check if step is first or last position,
+                           * in which case you should delete the step and
+                           * subsequently redirect the user to the step select
+                           * page for that position.
+                           */
+                          if (
+                            this.state.position ===
+                              getFirstPosition(integration, flowId) ||
+                            this.state.position ===
+                              getLastPosition(integration, flowId)
+                          ) {
+                            history.push(
```

Comment:
@gashcrumb @riccardo-forina - For some reason, when the redirect takes place it results in a blank page and gives the following errors: 
```
TypeError: Cannot read property 'description' of undefined toUIStep 
  87 | // this step is a Connection step
  88 | return {
  89 |   ...step,
> 90 |   description:
     | ^  91 |     (step as ConnectionOverview).description ||
  92 |     step.connection!.description ||
src/modules/integrations/components/editor/utils.ts:90
src/modules/integrations/components/editor/SelectConnectionPage.tsx:70
 70 |   steps: toUIStepCollection(integrationSteps),
```

It worked fine when it was just being done before confirming deletion from within the dialog (so, from line 117 or so). The difference seems to be that I'm using the "state"-defined position.  Any ideas on how I can get around this? Thanks!

Reply:
After some experimenting, I think it's not to do with the state-defined position, rather, the step itself. Looking at the Angular app doesn't seem to help a ton. Still trying out a few things.

Reply:
So thinking through this more, in the old editor the way I tackled this was to check if the step to be deleted was at the start or at the end of the step array, and if so instead of actually deleting the step, I'd simple overwrite it with an empty step (basically setting it as `{ id: key() }`.  So this may be the same approach we want to take here, as we're not really "adding" a step exactly, instead we're replacing the step with an unconfigured/empty one.